properties/base.py

```diff
@@ -296,15 +294,43 @@ def assert_valid(self, instance, value=None):
             value.validate()
         return True
 
+    def deserialize(self, value):
+        """Deserialize instance from JSON value
+
+        If a deserializer is registered, that is used. Otherwise, if the
+        instance_class is a HasProperties subclass, an instance can be
+        deserialized from a dictionary.
+        """
+        if self.deserializer:
+            return self.deserializer(value)
+        if value is None:
+            return None
+        if issubclass(self.instance_class, HasProperties):
+            return self.instance_class.deserialize(value)
+        return self.from_json(value)
+
     @staticmethod
-    def as_json(value):
-        """Serializes HasProperties instances to JSON"""
+    def to_json(value):
+        """Convert instance to JSON"""
         if isinstance(value, HasProperties):
-            return value.serialize(using='json')
-        elif value is None:
-            return None
-        else:
-            raise TypeError('Cannot serialize type {}'.format(value.__class__))
+            return value.serialize()
+        try:
+            return json.loads(json.dumps(value))
+        except TypeError:
+            raise TypeError(
+                'Cannot convert type {} to JSON without a calling it on '
+                'an Instance Property instance and registering a custom '
+                'serializer '.format(value.__class__.__name__)
+            )
```

Comment:
Hmm, I changed `it` to `serialize` to be a bit more explicit. Regarding "Instance Property instance" - it does sound strange, but I think it is correct. Maybe "instance of Instance Property" is slightly better?